compiler/evaltempl.nim

```diff
@@ -181,6 +182,7 @@ proc evalTemplate*(n: PNode, tmpl, genSymOwner: PSym;
   ctx.config = conf
   ctx.ic = ic
   initIdTable(ctx.mapping)
+  ctx.gensymHash = rand(0..int.high) #TODO: Wtf
```

Comment:
@Araq Do we already have a mechanism to get a hash/id thats unique per template invocation or something that could be used for this?

Reply:
didn't find anything either. You could add a counter to ``PContext``.

Reply:
Done, also had to pass it to the VM for `getAst` template expansion